index.rst

```diff
@@ -5,77 +5,331 @@
+It is also important that the results/outputs of a process follow a well defined format that can be consumed downstream.
+
+.. _introduction-process-image-data:
 
 Process Image Data
 ------------------
 
-Another important feature of the Catcher was the capability to process camera image data using the OCPS.
-The OCPS is a CSC that is capable of executing DM-pipeline processes.
+Another important feature of the ``Catcher`` is the capability to process camera image data using the ``OCPS``.
+The ``OCPS`` is a CSC that is capable of executing DM-pipeline processes.
 It is going to be key in allowing remote processing of Main Camera data during commissioning and operations.
 
-Nevertheless, if the Catcher is not a CSC it won't be able to interact direcly with OCPS to process image data.
-One alternative would be to interact with the OCPS remote backend direcly, instead of going through the CSC.
-For that, the OCPS backend implements a REST API that we could leverage to post requests and receive responses back.
+Since the ``Catcher`` is a CSC it can interact directly with ``OCPS`` to process image data, by sending commands to it.
+
+Alarms
+------
+
+Depending of the results of the analysis process the ``Catcher`` should be able to generate alerts to the system.
+For this we can follow a similar approach to that already implemented in the ``Watcher``.
+In short, the ``Catcher`` will have an "alarm" event with payloads to indicate the "severity" and other relevant information.
+There will also be commands to mute and acknowledge these alarms.
+We can also consider to borrow the alarm escalation model from the ``Watcher``.
 
 Generate reports
 ----------------
 
-The Catcher should also be able to generate reports, with graphs and performing additional lightweight computation.
-The original idea was to rely on parameterized Jupyter notebooks that would later be converted to webpages and rendered online so users could inspect the results, and use as reference for additional followup.
+The ``Catcher`` should also be able to generate reports, with graphs and additional lightweight computation.
+
+One of the initial ideas was to rely on parameterized Jupyter notebooks that would be processed using a `papermill`_-like application, converted to a webpage and rendered online so users could inspect the results, and use as reference for additional followup.
+
+.. _papermill: https://papermill.readthedocs.io/en/latest/
 
 This could still be achieved using a similar solution or relying on a service like noteburst :sqr:`065` :cite:`SQR-065` and Times Square :sqr:`066` :cite:`SQR-066` to execute parameterized notebooks using a web request and publishing the results afterwards.
 
+Another alternative is to implement a template system, where the user writes a template report with placeholders for text and images.
```

Comment:
I guess I don't quite understand how this differs from a parameterized notebook?

Reply:
I think in terms of unit testing and such this would be slightly easier to handle as we won't need to fiddle around with notebooks. But I think in terms of end products it won't be much different.